R/volcanoPlot.R

```diff
@@ -114,244 +108,381 @@ volcanoPlot <- function(contrastDF,
-    # plotType
-    if (plotType == "canvasXpress") {
-        symbolFill <- sapply(symbolFill, rgbaConversion, alpha = alpha, USE.NAMES = FALSE)
-
-        ## Create the canvasXpress cx.data and var.annot
-        cx.data <- data.frame(a = contrastDF[colnames(contrastDF) == x],
-                              b = contrastDF[colnames(contrastDF) == y])
-        colnames(cx.data) <- c(x, y)
-        var.annot <- data.frame(Group = contrastDF$group, LogInt = contrastDF$LogInt)
-        rownames(var.annot) <- rownames(cx.data)
-        events <- NULL
-
-        # Add geneSym labels in tooltips
-        if (!missing(geneSymCol)) {
-            var.annot <- cbind(var.annot, GeneLabel = contrastDF[[geneSymCol]])
-            events <- htmlwidgets::JS("{ 'mousemove' : function(o, e, t) {
-                                                if (o != null && o != false) {
-                                                    if (o.objectType == null) {
-                                                        t.showInfoSpan(e, '<b>' + o.y.vars + '</b> <br/>' +
-                                                        '<b>' + 'GeneLabel'  + '</b>' + ': ' + o.z.GeneLabel[0] + '<br/>' +
-                                                        '<b>' + o.y.smps[0]  + '</b>' + ': ' + o.y.data[0][0] + '<br/>' +
-                                                        '<b>' + o.y.smps[1]  + '</b>' + ': ' + o.y.data[0][1]);
-                                                    } else {
-                                                        t.showInfoSpan(e, o.display);
-                                                    };
-                                                }; }}")
-        }
+    if (any(is.null(symbolShape),
+            !is.character(symbolShape),
+            length(symbolShape)  != 3,
+            plotType == "canvasxpress" && !is.null(symbolShape) && length(.validate_cx_shapes(symbolShape)) != 3,
+            plotType == "ggplot" && !is.null(symbolShape) && length(.validate_gg_shapes(symbolShape)) != 3)) {
+        warning("symbolShape must be a vector of 3 charcter values. Assigning default values 'circle', 'circle', 'circle'.")
+        symbolShape  <- c("circle", "circle", "circle")
+
+    }
+
+    if (any(is.null(symbolColor),
+            !is.character(symbolColor),
+            length(symbolColor)  != 3,
+            length(.validate_colors(symbolColor)) != 3)) {
+        warning("symbolColor must be a vector of 3 character values. Assigning default values 'red3', 'grey25', 'deepskyblue4'.")
+        symbolColor <- c("red3", "grey25", "deepskyblue4")
+    }
+
+    if (any(is.null(transparency),
+            !is.numeric(transparency),
+            length(transparency) != 1,
+            transparency <= 0,
+            transparency > 1)) {
+        warning("transparency must be a singular value of class numeric and must be between 0 and 1. Assigning default value '0.5'.")
+        transparency <- 0.5
+    }
+
+    if (!is.null(referenceLine) &&
```

Comment:
I am not seeing a reference line in the older version. Where did this come from ?